Oracle.md

```diff
@@ -0,0 +1,162 @@
+
+### Range
+
+When an event has numerical outcomes that cannot be easily enumerated, they can be represented as a series with:
+
+- start: the first possible outcome number
+- count: the number of possible outcomes
+- step: the increment
+
+#### Example: tomorrow's temperature
+
+```
+start: -100
+count: 201
+step: 1
+```
+
+### Large range
+
+When a range of a numerical outcomes is large or unbounded, the oracle can represent the outcome using numerical decomposition.
+
+The event descriptor should include:
+
+- base: the base in which the outcome value is decomposed and which will indicate the possible range of each digit that will be signed.
+- is-signed: a boolean value indicating whether the outcomes can be negative.
+- r-values: an array of R values, one for each of the digit that will be signed by the oracle. If the is-signed value is true, an additional R-value must be provided as the first element in this array, which will be used to sign the string "+" in case of an outcome with a positive value and the string "-" in case of an outcome with a negative one.
+
+In the case where the number of digits that make up the outcome value exceeds the number of r-values that the oracle committed to, the oracle should sign the maximum possible value that he can attest to. 
+
+
+#### Example: BTC/USD rate
+
+The following example illustrates how numerical decomposition can help reduce the number of required CETs while covering a large range of possible outcomes.
+
+Alice and Bob wish to enter into a DLC at time `t` when BTC/USD ~= $10000 with a maturity time of `t+1`.
+The oracle provides 6 R-values, meaning that it can attest to outcomes in the range `[0,999999]`:
+
+```
+base: 10
+RValues: [R0, R1, R2, R3, R4, R5]
+```
+
+Alice and Bob create a DLC with "relevant" outcomes between $10000 and $10999.
+Below $10000 Alice gets everything, and above $10999 Bob gets everything.
+They use the `R0` and `R1` values to cover the cases where BTC/USD < $10000 using `R0=0` and `R1=0`.
+They use `R0`, `R1` and `R2` to cover the cases where BTC/USD > $10999 and <= $19999, using `R1=0`, `R1=1` and `R2 in [0-9]`.
+They use `R0` and `R1` to cover the cases where BTC/USD >= $20000 and < $99999, using `R0=0` and `R1 in [2,9]`.
+Finally, they use only `R0` to cover the cases where BTC/USD >= $100000 and <= $999999, using `R0 in [1-9]`.
+
+At maturity time, if the BTC/USD rate is less than $999999, the oracle sign each of the digit representing the outcome value.
+If the rate is greater than $999999, the oracle signs the value $999999.
+
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `external_event_descriptor`
+
+1. type: 55300 (`external_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
```

Comment:
Should we even include the nonce here? Otherwise it may be hard for them to do multiple nonces

Reply:
or maybe we should just do 
```
* [`u16`:`nb_nonces`]
* [`nb_nonces*x_point`:`oracle_nonces`]
```

Probably not a huge deal but might be worth doing

Reply:
I don't think it makes sense to add the nonces, since IIUC the point of this format is to provide oracle info through a different representation than TLV. If we do add nonces, it means one has to first parse the original representation, extract the nonces, put them in the good place and then add the original representation, it seems like it is defeating the purpose. But I think @LLFourn wanted to use that so maybe he should give us a hint on what would work best for him :)

Reply:
I think both removing the nonce or having multiple is reasonable, but I agree that we shouldn't specify just one.

Reply:
I hadn't noticed that I had actually added the nonce field 🤦 . Removed.